Dockerfile

```
@@ -501,6 +501,96 @@ RUN \
 
 # =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
+FROM sdk-libc as sdk-go-fips
+
+ARG ARCH
+ARG TARGET="${ARCH}-bottlerocket-linux-gnu"
+ARG SYSROOT="/${TARGET}/sys-root"
+ARG GOVER="1.21.4"
+ARG AWS_LC_FIPS_VER="2.0.2"
+
+USER root
+RUN DEBIAN_FRONTEND=noninteractive dnf install -y golang cmake xz wget ca-certificates clang python curl git
+
+USER builder
+WORKDIR /home/builder/sdk-go-fips
+COPY ./hashes/go /home/builder/hashes
+RUN \
+  sdk-fetch /home/builder/hashes && \
+  tar --strip-components=1 -xf go${GOVER}.src.tar.gz && \
+      rm go${GOVER}.src.tar.gz
+
+WORKDIR /home/builder/sdk-go-fips
+COPY patches/go-fips/* ./
+RUN git init && git apply --whitespace=nowarn *.patch
+
+# Build aws-lc crypto modules
+USER root
+WORKDIR /boring/boringssl
+COPY ./hashes/aws-lc /home/builder/hashes
+ENV UPSTREAM_SOURCE_FALLBACK=true
+RUN \
+  sdk-fetch /home/builder/hashes && \
+  tar --strip-components=1 -xf AWS-LC-FIPS-${AWS_LC_FIPS_VER}.tar.gz && \
+    rm AWS-LC-FIPS-${AWS_LC_FIPS_VER}.tar.gz
+RUN \
+  install -p -m 0644 -Dt /aws-lc/licenses LICENSE
+
+# Build BoringCrypto.
+WORKDIR /boring
+ARG LANG=C
+ARG ClangV=17
+ARG CFLAGS="-O2 -g -pipe -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions"
```

Does `clang` not support the standard CFLAGS?
```suggestion
ARG CFLAGS="-O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-clash-protection"
```

If we can, it'd be good to consolidate these steps into the same stage:
* `# Build aws-lc crypto modules`
* `# Build BoringCrypto.`
*` # Build Go BoringCrypto syso.`

Otherwise it's difficult to keep track of how the pieces fit together and what flags are defined where.

---

Clang doesn't use certain flags it seems (which results in error):
```bash
clang-17: compiler_error: argument unused during compilation: '-fstack-clash-protection' [-Werror,-Wunused-command-line-argument]
```

---

Specifying `-D_FORTIFY_SOURCE=2` for clang also gives me:
```bash
1.520 /aarch64-bottlerocket-linux-gnu/sys-root/usr/include/features.h:413:4: error: _FORTIFY_SOURCE requires compiling with optimization (-O) [-Werror,-W#warnings]
1.520   413 | #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
1.520       |    ^
1.520 1 error generated.
```

even though `-O2` is specified.

---

something is dropping the `-O2` flag to `-O0`:
```bash
1.520 clang version 17.0.4 (Fedora 17.0.4-1.fc39)
1.520 Target: aarch64-bottlerocket-linux-gnu
1.520 Thread model: posix
1.520 InstalledDir: /usr/bin
1.520 Selected GCC installation: /usr/lib/gcc/aarch64-bottlerocket-linux-gnu/11
1.520 Candidate multilib: .;@m64
1.520 Selected multilib: .;@m64
1.520  (in-process)
1.520  "/usr/bin/clang-17" -cc1 -triple aarch64-bottlerocket-linux-gnu -emit-obj -mrelax-all -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name jitterentropy-timer.c -mrelocation-model pic -pic-level 2 -fhalf-no-semantic-interposition -mframe-pointer=non-leaf -fmath-errno -ffp-contract=on -fno-rounding-math -mconstructor-aliases -funwind-tables=2 -target-cpu generic -target-feature +neon -target-feature +v8a -target-abi aapcs -debug-info-kind=constructor -dwarf-version=4 -debugger-tuning=gdb -v -fcoverage-compilation-dir=/boring/boringssl/build -resource-dir /usr/bin/../lib/clang/17 -dependency-file third_party/jitterentropy/CMakeFiles/jitterentropy.dir/jitterentropy-timer.c.o.d -MT third_party/jitterentropy/CMakeFiles/jitterentropy.dir/jitterentropy-timer.c.o -sys-header-deps -canonical-system-headers -D GOBORING -D FIPS_ENTROPY_SOURCE_PASSIVE -I /boring/boringssl/build/symbol_prefix_include -I /boring/boringssl/include -D AWS_LC_STDALIGN_AVAILABLE -D AWS_LC_BUILTIN_SWAP_SUPPORTED -D AWS_LC_URANDOM_U32 -D _XOPEN_SOURCE=700 -D NDEBUG -D AWSLC -D_FORTIFY_SOURCE=2 -D_GLIBCXX_ASSERTIONS -isysroot /aarch64-bottlerocket-linux-gnu/sys-root -internal-isystem /usr/bin/../lib/clang/17/include -internal-isystem /aarch64-bottlerocket-linux-gnu/sys-root/usr/local/include -internal-isystem /usr/lib/gcc/aarch64-bottlerocket-linux-gnu/11/../../../../aarch64-bottlerocket-linux-gnu/include -internal-externc-isystem /aarch64-bottlerocket-linux-gnu/sys-root/include -internal-externc-isystem /aarch64-bottlerocket-linux-gnu/sys-root/usr/include -O0 -Wformat -Werror=format-security -Wall -Wextra -Wno-unused-parameter -Werror -Wunused -Wcomment -Wchar-subscripts -Wuninitialized -Wshadow -Wwrite-strings -Wformat-security -Wunused-result -Wvla -Wtype-limits -Werror -Wformat=2 -Wsign-compare -Wmissing-field-initializers -Wwrite-strings -Wall -Wnewline-eof -Wstring-concatenation -Wimplicit-fallthrough -Wmissing-prototypes -Wold-style-definition -Wstrict-prototypes -Wshadow -Wcast-align -Wmissing-field-initializers -Wshadow -Wswitch-enum -Wextra -Wall -Wconversion -pedantic -std=c99 -fconst-strings -fdebug-compilation-dir=/boring/boringssl/build -ferror-limit 19 -fvisibility=hidden -fwrapv -fno-signed-char -fgnuc-version=4.2.1 -fcolor-diagnostics -target-feature +outline-atomics -target-feature -fmv -faddrsig -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o third_party/jitterentropy/CMakeFiles/jitterentropy.dir/jitterentropy-timer.c.o -x c /boring/boringssl/third_party/jitterentropy/jitterentropy-timer.c
```